packages/server/src/util/indicators/index.ts

```diff
@@ -0,0 +1,140 @@
+import { LiquidityBound } from '@sommelier/shared-types';
+import { DexTrade } from 'services/bitquery/generated-types';
+
+export function getAllIndicators(marketData: DexTrade[]): { [indicatorName: string]: LiquidityBound } {
+    return {
+        bollingerSMANormalBand: getSMABollingerBands(marketData),
+        bollingerSMAWideBand: getSMABollingerBands(marketData, 2.5),
+        bollingerEMANormalBand: getEMABollingerBands(marketData),
+        bollingerEMAWideBand: getEMABollingerBands(marketData, 2.5),
+        rsi: getRSILevels(marketData)
+    }
+}
+
+export function getEMABollingerBands(marketData: DexTrade[], numStdDevs = 2): LiquidityBound {
+    const sampleData = marketData[0];
+    const ma = getEMA(marketData);
+    const stddev = getStandardDeviation(marketData);
+    const boundTerm = numStdDevs * stddev;
+
+    return {
+        baseTokenId: sampleData.baseCurrency.address,
+        quoteTokenId: sampleData.quoteCurrency.address,
+        indicatorName: 'BollingerBand',
+        currentPrice: marketData[0].quotePrice,
+        upperBound: ma + boundTerm,
+        lowerBound: ma - boundTerm,
+    };
+}
+
+export function getSMABollingerBands(marketData: DexTrade[], numStdDevs = 2): LiquidityBound {
+    const sampleData = marketData[0];
+    const ma = getSMA(marketData);
+    const stddev = getStandardDeviation(marketData);
+    const boundTerm = numStdDevs * stddev;
+
+    return {
+        baseTokenId: sampleData.baseCurrency.address,
+        quoteTokenId: sampleData.quoteCurrency.address,
+        indicatorName: 'BollingerBand',
+        currentPrice: marketData[0].quotePrice,
+        upperBound: ma + boundTerm,
+        lowerBound: ma - boundTerm,
+    };
+}
+
+export function getRSILevels(marketData: DexTrade[]): LiquidityBound {
+    const sampleData = marketData[0];
+    const currentPrice = sampleData.quotePrice;
+    const last14Days = marketData.slice(0, 14);
+
+    const [pctGains, pctLosses] = last14Days.reduce((acc: number[][], ohlc: DexTrade, index: number) => {
+        if (index === 0) return acc;
+        const [pctGains, pctLosses] = acc;
+
+        const currentClosePrice = parseFloat(ohlc.close_price || '0');
```

Comment:
all the parsing makes you think we should be doing it right off the response. is there any reason to keep these as strings?

Reply:
No there isn't - but it comes back from bitquery as a string. I tried doing it in `bitquery/fetcher.ts` but then it didn't like how I wasn't respecting the `DexTrade` type (which expects it to be a string, because it's generated by the gql schema, which matches bitquery).

Is there a way I can coerce it at the bitquery level?

Reply:
You'd have to change the schema so prob not worth it. The other way to do it is to force it into a class like you did with Uniswap. Let's just leave it like this for now.